packages/cli/src/config/setup/Setup.handler.ts

```diff
@@ -0,0 +1,185 @@
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import {
+    type ICommandHandler,
+    type IConfigLayer,
+    type IHandlerParameters,
+    type IHandlerResponseApi,
+    type IProfileTypeConfiguration,
+    ImperativeConfig,
+    ImperativeError,
+    ProfileInfo,
+    TextUtils,
+} from "@zowe/imperative";
+import * as termkit from "terminal-kit";
+import {
+    AbstractConfigManager,
+    MESSAGE_TYPE,
+    type ProgressCallback,
+    type inputBoxOpts,
+    type qpItem,
+    type qpOpts,
+} from "zowe-native-proto-sdk";
+
+export default class ServerInstallHandler implements ICommandHandler {
+    public async process(params: IHandlerParameters): Promise<void> {
+        const profInfo = new ProfileInfo("zowe");
+        await profInfo.readProfilesFromDisk();
+        const cliPromptApi = new CliPromptApi(profInfo, params.response);
+        const profile = await cliPromptApi.promptForProfile();
+        if (profile) {
+            params.response.console.log(
+                `SSH Profile Validated: ${(profInfo.getTeamConfig().api.layers.find(profile.name) as IConfigLayer).path}`,
+            );
+        }
+    }
+}
+
+export class CliPromptApi extends AbstractConfigManager {
+    constructor(
+        mProfilesCache: ProfileInfo,
+        private mResponseApi: IHandlerResponseApi,
+    ) {
+        super(mProfilesCache);
+    }
+
+    private term = termkit.terminal;
+    protected showMessage(message: string, type: MESSAGE_TYPE): void {
+        switch (type) {
+            case MESSAGE_TYPE.INFORMATION:
+                this.mResponseApi.console.log(message);
+                break;
+            case MESSAGE_TYPE.WARNING:
+                this.mResponseApi.console.log(TextUtils.chalk.yellow(message));
+                break;
+            case MESSAGE_TYPE.ERROR:
+                this.mResponseApi.console.log(TextUtils.chalk.red(message));
+                break;
+            default:
+                throw new ImperativeError({ msg: "Unknown message type" });
+        }
+    }
+
+    protected async showInputBox(opts: inputBoxOpts): Promise<string | undefined> {
+        return await this.mResponseApi.console.prompt(`${opts.title}: `.replace("::", ":"), {
+            hideText: opts.password,
+        });
+    }
+
+    protected async withProgress<T>(message: string, task: (progress: ProgressCallback) => Promise<T>): Promise<T> {
+        return await task(() => {});
+    }
+
+    protected async showQuickPick(opts: qpOpts): Promise<string | undefined> {
+        return (await this.showCustomQuickPick(opts)).label;
+    }
+
+    protected async showCustomQuickPick(opts: qpOpts): Promise<qpItem | undefined> {
+        return new Promise<qpItem | undefined>((resolve) => {
+            this.term.grabInput(true);
+
+            // Prepare menu items
+            const menuItems = opts.items.map((item) =>
+                item.separator
+                    ? `${"─".repeat(10)}Migrate from SSH Config${"─".repeat(10)}`
```

Comment:
Why use `${"─".repeat(10)}` since it's more than 10 characters 🤔  Am kidding, this is a cool way to make the number of hyphens dynamic 😋